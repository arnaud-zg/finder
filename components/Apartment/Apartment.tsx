import { inlineBlock, padding } from 'csstips'
import { rem } from 'csx'
import { format } from 'date-fns'
import React from 'react'
import { style } from 'typestyle'
import { colors } from '../../constants/colors'
import { Paragraph } from '../Paragraph'
import { Title } from '../Title'

interface IProps {
  description: string
  identifier: string
  locationIndication: string
  price: number
  publishTime: string
}

const tagClassname = style(inlineBlock, padding(rem(0.25)), {
  backgroundColor: colors.gray.toHexString(),
  borderRadius: rem(0.25),
  color: colors.light.toHexString(),
  $nest: {
    '&:hover': {
      backgroundColor: colors.lightGray.toHexString(),
      color: colors.dark.toHexString(),
    },
  },
})

export class Apartment extends React.PureComponent<IProps> {
  render() {
    const {
      description,
      identifier,
      locationIndication,
      price,
      publishTime,
    } = this.props
    return (
      <div id={identifier}>
        <Title content={description} level={3} />
        <div
          className={style(inlineBlock, {
            textDecoration: 'underline',
          })}
        >
          <Paragraph content={locationIndication} />
        </div>
        <div className={tagClassname}>{price / 100} â‚¬</div>
        <div
          className={`${tagClassname} ${style({
            marginLeft: rem(0.25),
          })}`}
        >
          {format(new Date(publishTime), "dd/MM/yyyy 'at' HH:mm")}
        </div>
      </div>
    )
  }
}
