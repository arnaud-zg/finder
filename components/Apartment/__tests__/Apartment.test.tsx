import React from 'react'
import renderer from 'react-test-renderer'
import { Apartment } from '../Apartment'

describe('Apartment', () => {
  it('renders correctly', () => {
    const testRenderer = renderer.create(
      <Apartment
        description="Ground floor Studio apartment with Balcony !"
        identifier="1"
        locationIndication="Paris"
        price={75000}
        publishTime="2019-09-29T13:50:42.926Z"
      />
    )
    expect(testRenderer.toJSON()).toMatchSnapshot()
  })
})
