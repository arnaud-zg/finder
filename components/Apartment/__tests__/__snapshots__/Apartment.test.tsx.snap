// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Apartment renders correctly 1`] = `
<div
  id="1"
>
  <h3
    className="fhwo3dt"
  >
    Ground floor Studio apartment with Balcony !
  </h3>
  <div
    className="f4lw1zx"
  >
    <p
      className="f2qtt8h"
    >
      Paris
    </p>
  </div>
  <div
    className="f12o3wr"
  >
    750
     â‚¬
  </div>
  <div
    className="f12o3wr f1n0dvv8"
  >
    29/09/2019 at 15:50
  </div>
</div>
`;
